packages/hebao_v3/contracts/base/libwallet/WalletData.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2017 Loopring Technology Limited.
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+enum SigRequirement
+{
+    MAJORITY_OWNER_NOT_ALLOWED,
+    MAJORITY_OWNER_ALLOWED,
+    MAJORITY_OWNER_REQUIRED,
+    OWNER_OR_ANY_GUARDIAN,
+    ANY_GUARDIAN
+}
+
+struct Request
+{
+    address[] signers;
+    bytes[]   signatures;
+    uint      validUntil;
+    address   wallet;
+}
+
+// Optimized to fit into 64 bytes (2 slots)
+struct Quota
+{
+    uint128 currentQuota;
+    uint128 pendingQuota;
+    uint128 spentAmount;
+    uint64  spentTimestamp;
+    uint64  pendingUntil;
+}
+
+enum GuardianStatus
+{
+    REMOVE,    // Being removed or removed after validUntil timestamp
+    ADD        // Being added or added after validSince timestamp.
+}
+
+// Optimized to fit into 32 bytes (1 slot)
+struct Guardian
+{
+    address addr;
+    uint8   status;
+    uint64  timestamp; // validSince if status = ADD; validUntil if adding = REMOVE;
+}
+
+struct Wallet
```

Comment:
Will change if you really want to, but because it's used everywhere I kind of like it being short. It also follows the style of other structs like `Guardian`, `Quota`, `Approval`,... which also don't end with `Data` either so I thought why do it for `Wallet`?

Reply:
I'm OK with the existing naming.